src/migrations/helpers/get-deployed-contracts.ts

```diff
@@ -27,7 +34,20 @@ export async function getDeployedContracts(): Promise<DeployedContracts> {
     config.isMainnet() ||
     (config.isHardhat() && config.FORK_MAINNET)
   ) {
-    deployedAddresses = mainnetAddresses;
+
+    // Deploy contracts for Stark Proxy recovery.
+    const { starkProxyNewImpls } = await deployStarkProxyV2({
```

Comment:
Could you put this in `getDeployedContractsForTest` instead? I don't think this is logically the right place to put this since we wouldn't want to trigger a deployment during a call to get deployed contracts unless we were in test